### ISSUE TITLE
VM must consume nameservers from virt-launcher pod

### ISSUE THREAD

**Issue by cynepco3hahue** [kubevirt/kubevirt#781]
Currently, we add only google nameserver to VM and do not consume nameservers from virt-launcher pod
Pod:
```
# cat /etc/resolv.conf 
nameserver 192.168.121.81
search kubevirt-test-default.svc.cluster.local svc.cluster.local cluster.local
options ndots:5
```
VM:
```
$ cat /etc/resolv.conf
nameserver 0.0.0.0
nameserver 0.0.0.0
nameserver 0.0.255.255
nameserver 8.8.8.8
```

**Comment by cynepco3hahue** [comment]
FYI @vladikr

**Comment by cynepco3hahue** [comment]
`dnsReadConfig` from https://golang.org/src/net/dnsconfig_unix.go looks promising

**Comment by rmohr** [comment]
Let's also start tweaking the hostname and the subdomain of the pod which we create for the vm. See: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pods-hostname-and-subdomain-fields

We would set the pods `hostName` to the name of the vm and set a `subbomain` on the pod which we could simply call `virtualmachines` or `vms` or `vm`. Then we can simply set the hostname of the vm to the name of the vm. Further we will have reasonable dns entries which will point to our vms. For instance a FQDN like `myvm.virtualmachines.my-namespace.svc.cluster.local`.

Finally the dns server can offer the correct host name (and cloud-init metadata too).

**Comment by rmohr** [comment]
Also related to https://github.com/kubevirt/kubevirt/pull/775#discussion_r171485334

**Comment by fabiand** [comment]
@vladikr wasn't this fixed with #831?

**Comment by vladikr** [comment]
> @vladikr wasn't this fixed with #831?

@fabiand yes, I think it can be closed if there is nothing else to it.
#831 covered the nameservers and #856 added the search domains. - so we can close #841 as well.

**Comment by fabiand** [comment]
Thanks
